### PR TITLE
updates for julia 0.7

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Comment by Datseris** [comment]
Hello. Is there any update in making PkgBenchmark work for Julia 0.7? It would be a great help for package developers that want to move over.

**Comment by KristofferC** [comment]
Indeed, the Pkg system is now completely different and there isn't right now really a straight up replacement. Easiest would be to remove the `benchmark/REQUIRE` functionality.

**Comment by Datseris** [comment]
Things have changed quite a bit, and there is no depwarning for `Pkg.dir` anymore.

**Comment by KristofferC** [comment]
Hm, are you sure. There should be, heh.

**Comment by Datseris** [comment]
Oh damn you are right...
```
julia> using Pkg

julia> Pkg.dir("DynamicalBilliards")
┌ Warning: Pkg.dir is only kept for legacy CI script reasons
└ @ Pkg.API API.jl:432
"C:\\Users\\datseris\\.julia\\v0.6\\DynamicalBilliards\\"
```
because stuff like `Pkg.add` were not throwing deps, I extrapolated. Sorry!

**Comment by Datseris** [comment]
With the addition of `pathto`, can this be resolved now?

https://github.com/JuliaLang/julia/pull/28310

File: src/runbenchmark.jl [hunk 1]
```diff
@@ -150,8 +150,8 @@ end
 function _runbenchmark_local(file, output, tunefile, retune)
     # Loading
     include(file)
-    suite = if isdefined(Main, :SUITE)
-        Main.SUITE
+    suite = if @isdefined SUITE
```
Comment (tkf) [inline src/runbenchmark.jl]: Isn't it better to use `Baes.include(Main, file)` and stick with `isdefined(Main, :SUITE)` as before?  `include(file)` pulls user's code into `PkgBenchmark` namespace, right?  The worst case scenario would be that users monkey-patch `PkgBenchmark` without knowing it.

**Comment by platawiec** [comment]
I'm currently getting:
```
ERROR: MethodError: no method matching _runbenchmark(::String, ::String, ::BenchmarkConfig, ::Nothing; retune=false, custom_loadpath="")
```
It seems that `tunefile` is getting returned as `nothing`. Not sure if this is particular to my setup. I also noticed a depwarn during precompile on `utils.jl:25`, with a `try` statement without `catch` or `finally`

```
Stacktrace:
 [1] uv_error at ./libuv.jl:85 [inlined]
 [2] unlink(::String) at ./file.jl:743
 [3] #rm#9(::Bool, ::Bool, ::Function, ::String) at ./file.jl:253
 [4] #rm at ./none:0 [inlined]
 [5] _withtemp(::getfield(PkgBenchmark, Symbol("##22#26")){Bool,String}, ::String) at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/util.jl:25
 [6] #21 at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/runbenchmark.jl:84 [inlined]
 [7] _with_reqs(::getfield(PkgBenchmark, Symbol("##21#25")){Bool,String}, ::String, ::Function) at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/util.jl:7
 [8] (::getfield(PkgBenchmark, Symbol("#do_benchmark#24")){Bool,String,String})() at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/runbenchmark.jl:83
 [9] (::getfield(PkgBenchmark, Symbol("##52#53")){getfield(PkgBenchmark, Symbol("#do_benchmark#24")){Bool,String,String},String,String})(::LibGit2.GitRepo) at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/util.jl:38
 [10] transact(::getfield(PkgBenchmark, Symbol("##52#53")){getfield(PkgBenchmark, Symbol("#do_benchmark#24")){Bool,String,String},String,String}, ::LibGit2.GitRepo) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/LibGit2/src/LibGit2.jl:958
 [11] _withcommit(::getfield(PkgBenchmark, Symbol("#do_benchmark#24")){Bool,String,String}, ::LibGit2.GitRepo, ::String) at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/util.jl:34
 [12] #benchmarkpkg#20(::Nothing, ::Nothing, ::Bool, ::String, ::Function, ::String, ::String) at /Users/pawel/.julia/packages/PkgBenchmark/LEuLm/src/runbenchmark.jl:105
 [13] (::getfield(PkgBenchmark, Symbol("#kw##benchmarkpkg")))(::NamedTuple{(:retune,),Tuple{Bool}}, ::typeof(benchmarkpkg), ::String, ::String) at ./none:0
 [14] top-level scope at none:0
```

**Comment by KristofferC** [comment]
I updated this PR a bunch and it seems to work ok. Had to get rid of the custom benchmark dependencies but hopefully, we can get those back as the package manager matures. 

Still need to rewrite the tests which are kinda hard for this package.

**Comment by Datseris** [comment]
ping here, pretty please

**Comment by KristofferC** [comment]
Try this PR out and see if it works and give some feedback.

**Comment by Datseris** [comment]
I can't due to https://github.com/JuliaCI/PkgBenchmark.jl/issues/67 . Let's hope someone else does it :)

**Comment by Roger-luo** [comment]
@KristofferC So this PR works besides the tests?

**Comment by Roger-luo** [comment]
@KristofferC tried locally with [Yao.jl](https://github.com/QuantumBFS/Yao.jl), looks like working, but why CI is broken? need new tests?

**Pull request merged**